Network testing: run tests without exported state

gleb-urvanov opened the pull request:
This branch does not contain exported chainspec as part of CI workflow, nor runtime upgrade.

gleb-urvanov committed:
persistence service introduced

gleb-urvanov committed:
liter errors fixed

gleb-urvanov committed:
Merge branch 'iznik' into feature/lowdb-persistence

gleb-urvanov committed:
linter error fixed, conflicting rules disabled

gleb-urvanov committed:
Merge branch 'account-info-rename' of https://github.com/Lezek123/substrate-runtime-joystream into feature/lowdb-persistence

gleb-urvanov committed:
altered object creation to match new api

gleb-urvanov committed:
merged joystream-types-register-missing

gleb-urvanov committed:
Merge branch 'joystream-types-register-missing' of https://github.com/Lezek123/substrate-runtime-joystream into feature/lowdb-persistence

gleb-urvanov committed:
altering code to match new api

gleb-urvanov committed:
merged substrate_version_upgrade

gleb-urvanov committed:
made tests working with new API

gleb-urvanov committed:
prettier changes applied

gleb-urvanov committed:
package.json change reverted

gleb-urvanov committed:
code cleaning

gleb-urvanov committed:
members and council scenarion fixes

gleb-urvanov committed:
verbose fixture arguments

gleb-urvanov committed:
verbose result functions for fixtures

gleb-urvanov committed:
merged iznik

gleb-urvanov committed:
lockfile updated

gleb-urvanov committed:
prettier errors fixed

gleb-urvanov committed:
git actions added to run the tests

gleb-urvanov committed:
moved test runner to separate workflow triggered by label

gleb-urvanov committed:
commit to trigger the workflow

gleb-urvanov committed:
commit to test branch where workflow is triggered

gleb-urvanov committed:
altered trigger condition

gleb-urvanov committed:
debug output removed, manual dispatch added

gleb-urvanov committed:
added execution rights to create chainspec script

gleb-urvanov committed:
job condition modified

gleb-urvanov committed:
condition altered, debug added

gleb-urvanov committed:
added version for debug

gleb-urvanov committed:
set proper version for debug

gleb-urvanov committed:
workflow fixed

gleb-urvanov committed:
run condition altered

gleb-urvanov committed:
removed debug output

gleb-urvanov committed:
merged iznik

gleb-urvanov committed:
log output fixed

gleb-urvanov committed:
merged iznik

gleb-urvanov committed:
resolve json module option added

gleb-urvanov committed:
removed separate dockerfile for tesing, added testing export data

gleb-urvanov committed:
Merge branch 'chainspec-import-members-forum-versioned-store' of https://github.com/mnaamani/joystream into chainspec-import-members-forum-versioned-store

gleb-urvanov committed:
added playbook for testing with exported state

gleb-urvanov committed:
changed playbook for testing workflow

gleb-urvanov committed:
chainspec renamed

gleb-urvanov committed:
worker payout test fixed

gleb-urvanov committed:
returned upgrade runtime to the testset

gleb-urvanov committed:
duration increased

gleb-urvanov committed:
runtime upgrade temporary removed

gleb-urvanov committed:
added console output for ci flow

gleb-urvanov committed:
worker payout fixed

gleb-urvanov committed:
merged iznik

mnaamani commented:
Could you fix the linter errors.

Also when I run yarn its getting modified with:

```
MokoBook:joystream mokhtar$ git diff
diff --git a/yarn.lock b/yarn.lock
index 33280ea44..57a1b8b00 100644
--- a/yarn.lock
+++ b/yarn.lock
@@ -3464,7 +3464,7 @@
   dependencies:
     "@babel/runtime" "^7.10.5"
 
-"@polkadot/keyring@^1.7.0-beta.5", "@polkadot/keyring@^3.0.1":
+"@polkadot/keyring@3.0.1", "@polkadot/keyring@^1.7.0-beta.5", "@polkadot/keyring@^3.0.1":
   version "3.0.1"
   resolved "https://registry.yarnpkg.com/@polkadot/keyring/-/keyring-3.0.1.tgz#3944079697c15b2af81e1f57b1c4aeab703a4fef"
   integrity sha512-vAHSBnisiDYHsBbEzAgIpuwQp3vIDN2uWQ/1wAE2BrKzXCBQM7RrF3LRcLFySk0xzQoDs7AP1TlPoakxJ/C/Qw==
```

So perhaps you can update it and check it in also.

mnaamani commented on tests/network-tests/package.json:
```diff
@@ -4,10 +4,10 @@
   "scripts": {
     "build": "tsc --build tsconfig.json",
-    "test": "tap --files src/iznik/tests/proposals/*Test.ts --files src/iznik/tests/workingGroup/*Test.ts -T",
+    "test": "tap --files src/iznik/tests/unknown.unknown src/iznik/tests/councilSetup.ts src/iznik/tests/proposals/*Test.ts src/iznik/tests/leaderSetup.ts src/iznik/tests/workingGroup/*Test.ts src/iznik/tests/proposals/updateRuntime.ts -T",
```
What is `src/iznik/tests/unknown.unknown` ?

gleb-urvanov replied:
That is a workaround for `node-tap` `--files` argument issue: the first file in the list will not be processed.

mnaamani commented:
For sanity check I ran `yarn build` in `tests/network-tests` and I get 76 build errors.
Is that a problem?

gleb-urvanov committed:
altered path to augment types

gleb-urvanov committed:
linter fixed

gleb-urvanov committed:
yarn.lock updated

gleb-urvanov committed:
typo fixed

gleb-urvanov committed:
nonce clearing for transaction expected to failure added

gleb-urvanov committed:
reverted nonce changes

gleb-urvanov committed:
Merge branch 'iznik' into feature/devops-update

gleb-urvanov committed:
removed docker-compose dependency for testing purposes

gleb-urvanov committed:
typo fixed

gleb-urvanov committed:
reward relationship for leader issue fixed

gleb-urvanov committed:
Bug/worker payout (#9)

* async fixture running issue fix

* scenario names fixed

* increased mint capacity increment

gleb-urvanov committed:
chainspec and runtime upgrade removal (#11)

Chainspec and runtime upgrade removed, also small fixes and improvements

gleb-urvanov committed:
Execution time improvement (#14)

fixed election parameters tests to work with altered block generation time, new chainspec built but removed from CI

gleb-urvanov committed:
added execute permission for new script
